src/read.lisp

```diff
@@ -547,10 +547,16 @@
           (skip-whitespaces-and-comments stream)
           (let ((ch (%peek-char stream)))
             (cond
-              ((or (null ch) (char= ch #\)))
+              ((null ch)
                (if eof-error-p
                    (error "End of file")
                    eof-value))
+              ((char= ch #\))
+               (if eof-error-p
+                   (progn 
+                     (%read-char stream)
+                     (error "End of file"))
```

Comment:
SBCL signals `unmatched close parenthesis` here instead of end-of-file.  Can we do the same?

Reply:
I guess you want to read the character when when eof-error-p is false, right?  You can do right before the conditional.

Reply:
Agreed on both points

Reply:
It looks like reading when eof-error-p is false messes up %read-list which is expecting to discard a #\) before finishing. What do you suggest here?
I've changed the error message.